### PR TITLE
set dirty when componentWillReceiveProps

### PR DESCRIPTION
**Pull request by wowlusitong**

### CODE REVIEW

**Comment by wowlusitong** [comment]
@tsing review please

File: src/State.js [hunk 1]
```diff
@@ -210,15 +210,17 @@ Object.defineProperty(State.prototype, 'dirty', {
...
+var makeSchema = function(schema, defaultValue, isDirty) {
   var validation = {};
   var value = {};
 
   schema = schema.withMutations(function(s) {
     s.keySeq().toArray().forEach(k => {
       var property = makeProperty(s.get(k));
       s.set(k, property);
-      value[k] = defaultValue[k] || s.get(k).props.get('defaultValue', null);
+      value[k] = isDirty
```
Comment (jqwn) [inline src/State.js]: isDirty 是否应该每个字段单独判断呢？

**Comment by wowlusitong** [comment]
@tsing review please

**Comment by jqwn** [comment]
LGTM

**Pull request merged**